server/node/src/main/java/org/kaaproject/kaa/server/operations/service/akka/actors/core/ApplicationLogActorMessageProcessor.java

```diff
@@ -97,8 +97,12 @@ protected void processLogEventPack(ActorContext context, LogEventPackMessage mes
         List<LogAppender> required = filterAppenders(logSchema.getVersion(), true);
         List<LogAppender> optional = filterAppenders(logSchema.getVersion(), false);
         if (required.size() > 0 || optional.size() > 0) {
-            for (LogAppender logAppender : optional) {
-                logAppender.doAppend(message.getLogEventPack(), voidCallback);
+            if (required.size() > 0) {
```

Comment:
This is logically wrong. We can't replace required with optional. Optional appenders should not affect data delivery. You need to change code that is related to required appenders processing.